src/auth/auth.cpp

```diff
@@ -535,32 +497,66 @@ std::vector<auth::User> Auth::AllUsersForRole(const std::string &rolename_orig)
 }
 
 #ifdef MG_ENTERPRISE
-bool Auth::GrantDatabaseToUser(const std::string &db, const std::string &name, system::Transaction *system_tx) {
+bool Auth::GrantDatabase(const std::string &db, const std::string &name, system::Transaction *system_tx) {
```

Comment:
When auth module used we need to be more explicit about the failure reason.
Normally its just false -> no user or role with that name
With auth module false must be -> no role with that name

Change return type to enum + better diagnostic in interpreter